targets/esp32/mqtt/src/main.c

```diff
@@ -0,0 +1,131 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stddef.h>
+#include <string.h>
+#include "esp_wifi.h"
+#include "esp_system.h"
+#include "nvs_flash.h"
+#include "esp_event.h"
+#include "esp_netif.h"
+
+#include "freertos/FreeRTOS.h"
+#include "freertos/task.h"
+#include "freertos/semphr.h"
+#include "freertos/queue.h"
+
+#include "lwip/sockets.h"
+#include "lwip/dns.h"
+#include "lwip/netdb.h"
+
+#include "esp_log.h"
+#include "mqtt_client.h"
+#include "cnetwork.h"
+#include "config.h"
+
+#define CLIENT_ID "ESP32"
+
+static const char *TAG = "MQTT_MAINFLUX";
+
+static void log_error_if_nonzero(const char *message, int error_code)
+{
+    if (error_code != 0)
+    {
+        ESP_LOGE(TAG, "Last error %s: 0x%x", message, error_code);
+    }
+}
+void create_mainflux_topic(void)
+{
+    const char *_preId = "channels/";
+    const char *_postId = "/messages";
+    strcpy(mfTopic, _preId);
+    strcat(mfTopic, mfChannelId);
+    strcat(mfTopic, _postId);
+}
+
+static void mqtt_event_handler(void *handler_args, esp_event_base_t base, int32_t event_id, void *event_data)
+{
+    ESP_LOGD(TAG, "Event dispatched from event loop base=%s, event_id=%" PRIi32 "", base, event_id);
+    esp_mqtt_event_handle_t event = event_data;
+    esp_mqtt_client_handle_t client = event->client;
+    int msg_id;
+    create_mainflux_topic();
+    switch ((esp_mqtt_event_id_t)event_id)
+    {
+    case MQTT_EVENT_CONNECTED:
+        ESP_LOGI(TAG, "MQTT_EVENT_CONNECTED");
+        msg_id = esp_mqtt_client_publish(client, mfTopic, "'{'message':'hello mainflux'}", 0, 0, 0);
```

Comment:
Why do you publish here?

Reply:
Done on connection event to demonstrate connectivity